src/components/changelog/ChangelogWrapper.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import NavFrontendModal from 'nav-frontend-modal';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchChangelogs } from '../../store/changelog/changelog_actions';
+import ChangelogModal from '../../components/changelog/ChangelogModal';
+import { ApplicationState, store } from '../../store';
+import { Changelog } from '../../store/changelog/changelogTypes';
+
+NavFrontendModal.setAppElement('#maincontent');
+const getPropsFromState = (state: ApplicationState) => ({
+    changelogs: state.changelogs.data,
+});
+
+interface ChangelogStorage {
+    viewed_version: number;
+}
+
+const CHANGELOG_LOCAL_KEY = 'SYFOOVERSIKT_CHANGELOG';
+
+const getChangelogStorage = (): ChangelogStorage | undefined => {
+    const stored = localStorage.getItem(CHANGELOG_LOCAL_KEY);
+    if (stored !== null) {
+        return JSON.parse(stored);
+    }
+    return undefined;
+};
+
+const createChangelogStorage = (version: number) => ({
+    viewed_version: version,
+}) as ChangelogStorage;
+
+const saveChangelogVersionViewed = (version: number = 0) => {
+    localStorage.setItem(CHANGELOG_LOCAL_KEY, JSON.stringify(createChangelogStorage(version)));
+};
+
+export default () => {
+
+    const dispatch = useDispatch();
+    const [showChangelog, setShowChangelog] = useState(false);
+    const [latestChangelog, setLatestChangelog] = useState<Changelog | undefined>(undefined);
+
+    useEffect(() => {
+        dispatch(fetchChangelogs());
+    }, []);
+
+    const {
+        changelogs,
+    } = useSelector(getPropsFromState);
+
+    useEffect(() => {
+        if (changelogs.length > 0) {
+            const lastChangelog = [...changelogs].sort((a, b) => {
+                return a.version > b.version
+                    ? -1
+                    : 1;
+            })[0];
+
+            if (lastChangelog === undefined) return;
+
+            const storedSettings = getChangelogStorage();
+            if (storedSettings) {
```

Comment:
Enkel If i stedet for nested if?

Reply:
Har ryddet litt nå. Litt usikker på hva du tenker ang egen funksjon. Tar gjerne imot forslag.